pkg/deploy/api/validation/validation.go

```diff
@@ -50,6 +53,60 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	return allErrs
 }
 
+func getContainerImageNames(template *kapi.PodTemplateSpec) []string {
+	originalContainerImageNames := make([]string, 0)
+	for _, container := range template.Spec.Containers {
```

Comment:
for i := range template.Spec.Containers {
  originalContainerImageNames[i] = template.Spec.Containers[i].Image
}